### DISCUSSION TITLE
wasm http-request doesn't return

### DISCUSSION THREAD

**Discussion by wgtun** [Desdaemon/flutter_rust_bridge_template#50] [Q&A] [answered]
Hello, I would like to ask a question, I added a http request code, the mobile is normal, but the web does not return, I want to know where is my problem, web http request code like this:

```
use gloo_net::http::{Headers, Request};
#[tokio::main(flavor = "current_thread")]
pub fn request(params: MyRequest) -> Result<String> {
            let headers = get_web_default_headers().unwrap();
            for header in &params.headers {
                headers.set(&header.key, &header.value);
            }
            let result = Request::get(&params.url).headers(headers).send().await?;
            Ok(result.text().await?)
}
```

**Comment by Desdaemon** [comment, marked as answer]
You'll need to use `#[cfg]` attributes to enable Tokio only for non-web users and [`wasm-bindgen-futures`](https://lib.rs/crates/wasm-bindgen-futures) for web users. Also I'll open discussions so common questions like this can be collected in one place.
